packages/react-openapi/src/OpenAPIServerURLForm.tsx

```diff
@@ -0,0 +1,63 @@
+'use client';
+
+import * as React from 'react';
+import { useRouter } from 'next/navigation';
```

Comment:
We can't use `next` modules in `react-openapi` as the main application doesn't use `next`.

We need to find a way to pass these by props to the high level component.

Or to use a client context:

```
import { OpenAPI, OpenAPIContext } from '@gitbook/react-openapi';

return (
  <OpenAPIContext onUpdateEnums={} enums={..}>
)
```

Reply:
This is updated to use a context. 

I have one problem that I'm not sure how best to resolve though - using a client component (with useContext) means I can't update the url in the Request code block as that is a server component. And I don't want to bring the codeblock code onto the client.

Looking at whether I can use `nuq/server` to cache the url in query param atm.